src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.ts

```diff
@@ -968,6 +968,134 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/infinite-hi
+      const helper = algoliasearchHelper(createSearchClient(), 'indexName', {
+        index: 'indexName',
+      });
+
+      const renderState1 = infiniteHitsWidget.getRenderState!(
+        {},
+        createInitOptions({ state: helper.state, helper })
+      );
+
+      expect(renderState1.infiniteHits).toEqual({
+        hits: [],
+        sendEvent: expect.any(Function),
+        bindEvent: expect.any(Function),
+        isFirstPage: true,
+        isLastPage: true,
+        results: undefined,
+        showMore: expect.any(Function),
+        showPrevious: expect.any(Function),
+        widgetParams: {},
+      });
+
+      const hits = [
+        { objectID: '1', name: 'name 1' },
+        { objectID: '2', name: 'name 2' },
+      ];
+
+      const results = new SearchResults(helper.state, [
+        createSingleSearchResponse({ hits, queryID: 'theQueryID' }),
+      ]);
+
+      const renderState2 = infiniteHitsWidget.getRenderState!(
+        {},
+        createRenderOptions({
+          helper,
+          state: helper.state,
+          results,
+        })
+      );
+
+      const expectedHits = [
+        { objectID: '1', name: 'name 1', __queryID: 'theQueryID' },
+        { objectID: '2', name: 'name 2', __queryID: 'theQueryID' },
+      ];
+
+      expect(renderState2.infiniteHits).toEqual({
+        hits: expectedHits,
+        sendEvent: renderState1.infiniteHits!.sendEvent,
+        bindEvent: renderState1.infiniteHits!.bindEvent,
+        isFirstPage: true,
+        isLastPage: true,
+        results,
+        showMore: renderState1.infiniteHits!.showMore,
+        showPrevious: renderState1.infiniteHits!.showPrevious,
+        widgetParams: {},
+      });
+    });
+  });
+
+  describe('getWidgetRenderState', () => {
+    it('returns the widget render state', () => {
+      const renderFn = jest.fn();
+      const unmountFn = jest.fn();
+      const createInfiniteHits = connectInfiniteHits(renderFn, unmountFn);
+      const infiniteHitsWidget = createInfiniteHits({});
+      const helper = algoliasearchHelper(createSearchClient(), 'indexName', {
+        index: 'indexName',
+      });
+
+      const renderState1 = infiniteHitsWidget.getWidgetRenderState!(
+        createInitOptions({ state: helper.state, helper })
+      );
+
+      expect(renderState1).toEqual({
+        hits: [],
+        sendEvent: expect.any(Function),
+        bindEvent: expect.any(Function),
+        isFirstPage: true,
+        isLastPage: true,
+        results: undefined,
+        showMore: expect.any(Function),
+        showPrevious: expect.any(Function),
+        widgetParams: {},
+      });
+
+      const hits = [
+        { objectID: '1', name: 'name 1' },
+        { objectID: '2', name: 'name 2' },
+      ];
+
+      const results = new SearchResults(helper.state, [
+        createSingleSearchResponse({ hits, queryID: 'theQueryID' }),
+      ]);
+
+      const renderState2 = infiniteHitsWidget.getWidgetRenderState!(
```

Comment:
same here, one test for init, one for render will be clearer if we ever break anything on it